test/segment-loader.test.js

```diff
@@ -20,6 +23,75 @@ const ogAddSegmentMetadataCue_ = SegmentLoader.prototype.addSegmentMetadataCue_;
 
 SegmentLoader.prototype.addSegmentMetadataCue_ = function() {};
 
+QUnit.module('SegmentLoader Isolated Functions');
+
+QUnit.test('illegalMediaSwitch detects illegal media switches', function(assert) {
+  let startingMedia = { containsAudio: true, containsVideo: true };
+  let newSegmentMedia = { containsAudio: true, containsVideo: true };
+
+  assert.notOk(illegalMediaSwitch('main', startingMedia, newSegmentMedia),
+               'no error when muxed to muxed');
+
+  startingMedia = { containsAudio: true, containsVideo: false };
+  newSegmentMedia = { containsAudio: true, containsVideo: false };
+  assert.notOk(illegalMediaSwitch('main', startingMedia, newSegmentMedia),
+               'no error when audio only to audio only');
+
+  startingMedia = { containsAudio: false, containsVideo: true };
+  newSegmentMedia = { containsAudio: false, containsVideo: true };
+  assert.notOk(illegalMediaSwitch('main', startingMedia, newSegmentMedia),
+               'no error when video only to video only');
+
+  startingMedia = { containsAudio: false, containsVideo: true };
+  newSegmentMedia = { containsAudio: true, containsVideo: true };
+  assert.notOk(illegalMediaSwitch('main', startingMedia, newSegmentMedia),
+               'no error when video only to muxed');
+
+  startingMedia = { containsAudio: true, containsVideo: false };
+  newSegmentMedia = { containsAudio: true, containsVideo: true };
+  assert.equal(illegalMediaSwitch('main', startingMedia, newSegmentMedia),
+               'Video found in segment when we expected only audio.' +
+               ' We can\'t switch to a stream with video from a stream that only had' +
+               ' audio.' +
+               ' To get rid of this message, please add codec information to the' +
+               ' manifest.',
+               'error when audio only to muxed');
+
+  startingMedia = { containsAudio: true, containsVideo: true };
+  newSegmentMedia = { containsAudio: true, containsVideo: false };
+  assert.equal(illegalMediaSwitch('main', startingMedia, newSegmentMedia),
+               'Only audio found in segment when we expected video.' +
+               ' We can\'t switch to audio only from a stream that had video.' +
+               ' To get rid of this message, please add codec information to the' +
+               ' manifest.',
+               'error when muxed to audio only');
+
+  startingMedia = { containsAudio: true, containsVideo: true };
```

Comment:
Maybe add a test that has `startingMedia = { containsAudio: true, containsVideo: false };` and `newSegmentMedia = { containsAudio: false, containsVideo: false };` just to confirm it also returns `'Neither audio nor video found in segment.'`

Reply:
Looks like you forgot to updated `containsVideo` here, this test is exactly the same as the one on line 55. Also would be nice to move the two tests for `'Neither audio nor video found in segment.'` next to each other just for reading through, but not necessary